packages/contracts-bedrock/test/properties/halmos/OptimismSuperchainERC20.t.sol

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import { OptimismSuperchainERC20 } from "src/L2/OptimismSuperchainERC20.sol";
+import { SymTest } from "halmos-cheatcodes/src/SymTest.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { ERC1967Proxy } from "@openzeppelin/contracts-v5/proxy/ERC1967/ERC1967Proxy.sol";
+import { MockL2ToL2Messenger } from "./MockL2ToL2Messenger.sol";
+import { AdvancedTest } from "../helpers/AdvancedTests.sol";
+
+contract SymTest_OptimismSuperchainERC20 is SymTest, AdvancedTest {
+    MockL2ToL2Messenger internal constant MESSENGER = MockL2ToL2Messenger(Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER);
+
+    OptimismSuperchainERC20 public superchainERC20Impl;
+    OptimismSuperchainERC20 internal optimismSuperchainERC20;
+
+    function setUp() public {
+        // Deploy the OptimismSuperchainERC20 contract implementation and the proxy to be used
+        superchainERC20Impl = new OptimismSuperchainERC20();
+        optimismSuperchainERC20 = OptimismSuperchainERC20(
+            address(
+                // TODO: Update to beacon proxy
+                new ERC1967Proxy(
+                    address(superchainERC20Impl),
+                    abi.encodeCall(OptimismSuperchainERC20.initialize, (remoteToken, name, symbol, decimals))
+                )
+            )
+        );
+
+        // Etch the mocked L2 to L2 Messenger since the messenger logic is out of scope for these test suite. Also, we
+        // avoid issues such as `TSTORE` opcode not being supported, or issues with `encodeVersionedNonce()`
```

Comment:
didn't halmos' nightly build include support for tstore already? :thinking: or am I mixing it up with something else?